### PR TITLE
Gateway overview scaffolding

### PR DESCRIPTION
**Pull request by PavelJankoski**
#### Summary

Gateway overview scaffolding and new gateway overview header component.

#### Changes

- Refactor gateway connection component.
- Add placeholder divs for gateway overview panels.
- Create gateway overview header component.
- Add/Remove gateway bookmark.

##### Steps

1. Login to console.
2. Navigate to gateways page.
3. Select gateway.
4. Check the gateway overview page.

##### Results

image

#### Notes for Reviewers

This is still work in progress. Menu dropdown items should be updated in the header and small screen behaviour to be defined.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+.root
+  position: sticky
+  top: 0
+  z-index: calc($zi.dropdown - 1)
+  display: flex
+  justify-content: space-between
+  gap: $cs.xl
+  align-items: center
+  padding: $cs.m
+  background-color: $c['tts-primary-050']
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl]:
```suggestion
  background-color: var(--c-bg-neutral-min)
```
Let's use this actually. I've updated the design as well.

**Review by kschiffer — COMMENTED** [review]
So far LGTM

File: pkg/webui/console/containers/gateway-overview-header/index.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        type: toast.types.ERROR,
+      })
+    }
+  }, [gateway_id])
+
+  const menuDropdownItems = (
+    <>
+      <Dropdown.Item title={sharedMessages.downloadGlobalConf} action={handleGlobalConfDownload} />
+      <Dropdown.Item title={m.duplicateGateway} action={() => {}} />
+      <Dropdown.Item title={sharedMessages.deleteGateway} action={() => {}} />
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-overview-header/index.js]: You should be able to connect that already. After deletion the user should be sent back to the gateway list with a toast displaying.

File: pkg/webui/console/containers/gateway-overview-header/index.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        message: sharedMessages.globalConfFailedMessage,
+        type: toast.types.ERROR,
+      })
+    }
+  }, [gateway_id])
+
+  const menuDropdownItems = (
+    <>
+      <Dropdown.Item title={sharedMessages.downloadGlobalConf} action={handleGlobalConfDownload} />
+      <Dropdown.Item title={m.duplicateGateway} action={() => {}} />
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-overview-header/index.js]: Let's comment this one out for now.

File: pkg/webui/console/views/gateway-overview/index.js [hunk 4]
```diff
@@ -152,24 +151,21 @@ const GatewayOverview = () => {
...
+        <div className="item-12 md:item-12 lg:item-6 sm:item-6">
+          <div style={{ height: '30rem', backgroundColor: 'lightgray' }} />
         </div>
-        <div className="item-12 md:item-12 lg:item-6 sm:item-6" />
         <div className="item-12 md:item-12 lg:item-6 sm:item-6">
           <BlurryNetworkActivityPanel />
         </div>
+        <div className="item-12 md:item-12 lg:item-6 sm:item-6">
+          <div style={{ height: '30rem', backgroundColor: 'lightgray' }} />
+        </div>
```
Comment (kschiffer) [inline pkg/webui/console/views/gateway-overview/index.js]: You can already include the gateway status and map here.

**Review by kschiffer — CHANGES_REQUESTED** [review]
Thanks, just some small remarks.

**Comment by PavelJankoski** [comment]
@kschiffer Can you do one last review before merging? With the latest delete gateway integration, the component become a little bit crowded with all the deleting logic (about purging and stuff). I would appreciate if you have any suggestion to improve it or be able to use the already created `delete-modal-button` component.

File: pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl [hunk 5]
```diff
@@ -29,12 +29,12 @@
     white-space: nowrap
     color: var(--c-text-neutral-heavy)
     font-weight: $fw.bold
-    font-size: $fsv2.m
+    font-size: 1.15rem
```
Comment (kschiffer) [inline pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl]: Where's that size coming from?

**Review by kschiffer — APPROVED** [review]

File: pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl (hunk 5 above)
Comment (PavelJankoski) [reply pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl]: @kschiffer I added that regarding the discussion on Slack. I cannot find a font-size matching `16px`, should we add one?
image

Comment (kschiffer) [reply pkg/webui/console/containers/gateway-overview-header/gateway-overview-header.styl]: Ah ok, let's use `$fsv2.l` then. I've updated the designs as well.

**Pull request merged**